torch/_guards.py

```diff
@@ -763,6 +763,32 @@ def current_trace_id():
         return TraceId(self.compile_id, self.attempt)
 
 
+class RuntimeCompileContext:
+    """
+    For tracking compile-time information that we want available at runtime,
+    e.g., to allow us to query and log at runtime the CompileId active when
+    a module was compiled. A RuntimeCompileContext and a CompileContext
+    should never be active at the same time.
+    """
+
+    @staticmethod
+    def try_get() -> Optional[RuntimeCompileContext]:
+        return getattr(_TLS, "runtime_compile_context", None)
```

Comment:
Would `runtime_compile_context` and `compile_context` ever be set at the same time?  What would that mean?  There also seems like there is overlap with TracingContext below.

If all three of these can be set at the same time, would the trace_id differ?

I'm not totally convinced we need all 3.  Maybe we could have a single context, but with subclasses with more or less information.   That would also simplify lookups.